### PR TITLE
Group By

### PR DESCRIPTION
**Pull request by curiousest**

### CODE REVIEW

File: pywesome/pywesome.py [hunk 1]
```diff
@@ -136,6 +136,33 @@ def sum(collection, prop=None):
...
+def group_by(collection, key, key_is_unique=False, 
+             key_transformation_function=lambda a: a, 
+             get_method=getattr):
+        '''
+        :param key: the key to group entities in the collection by
+        :param key_is_unique: whether or not the key is unique. If False, the
+        returning dictionary will contain lists of entities instead of entities
+        :param key_transformation_function: a method used to transform the input
+        keys to be used in the dictionary. Default is to just use the key.
+        :param get_method: the method used to retrieve the field from the item
```
Comment (curiousest) [inline pywesome/pywesome.py:157]: This is so we can support both things like:
`entity.x` and `entity['x']`

**Comment by lucasbedout** [comment]
Please add the tests 👍

**Comment by curiousest** [comment]
Ok now? It's very... pythonic...

**Comment by curiousest** [comment]
*bump*

**Comment by curiousest** [comment]
*bump*

**Comment by lucasbedout** [comment]
Seems really too complex / overkilled to me but if you need it go on, sorry I thought I approved 3 days ago

**Comment by curiousest** [comment]
The apparent complexity is mostly in the value transformation/getattr function parameters. Without those, it'd look much much simpler. They cause almost every value access/usage/declaration to be wrapped in a function.

**Comment by curiousest** [comment]
When this happens in practice, do you think it's better to have two functions:
One with the simple implementation and simple parameters and one with more complex options.
OR one:
The same complex function that works for both?

**Pull request merged**